scripts/artifacts-building/README.md

```diff
@@ -0,0 +1,115 @@
+   the repo server is built) using the standard repo-build playbook from
+   OSA, but with selective git cloning, selective wheel building and
+   selective venv building disabled so that we have packages and venvs
+   for everything supported by RPC-O. Published artifacts can be found
+   on [rpc-repo](http://rpc-repo.rackspace.com/os-releases/). The job
+   executes ``scripts/artifacts-building/python/build-python-artifacts.sh``.
+
+4. Container artifacts are built using the previously built apt and python
+   artifacts. This is done by preparing the host as a LXC host, then building
+   a series on containers in turn without starting them but instead executing
+   the appropriate roles against them using a chroot connection and tagfilter
+   strategy. The tagfilter strategy ignores all ``-config`` tasks and filters
+   out all notifiers so that the chroot only has the right apt packages
+   installed and the right python packages and venvs installed. Published
+   artifacts can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/lxc-images/). The job executes
+   ``scripts/artifacts-building/containers/build-process.sh``.
+
+As RPC-O executes deploy tests in PR's where the value for ``rpc_release`` is
+being changed in the PR, the artifact build scripts have been designed to
+adapt based on whether the previous artifacts exist for the given release or
+not. The PR tests are also implemented in parallel so that they give results
+back quickly, whereas the periodic jobs are set to run in the proper sequence.
+The PR jobs are all read-only (they do not upload to rpc-repo), but the
+periodic jobs will upload to rpc-repo if artifacts for the given release do
+not exist yet. To replace artifacts for a series, execute the periodic job
+for the series with ``REPLACE_ARTIFACTS`` set to ``YES``.
+
+#### Artifact Consumption Process
+
+Artifacts are consumed by implementing the appropriate variable overrides for
+OSA to consume our artifacts instead of using the default upstream artifacts.
+
+* Apt artifact consumption configuration is implemented in
+  ``group_vars/all/apt.yml`` and in the lower part of
+  ``rpcd/etc/openstack_deploy/user_rpco_variables_defaults.yml``.
+
+* Git and python artifacts are staged using
+  ``rpcd/playbooks/stage-python-artifacts.yml``, then consumed as they are in
+  any normal RPC-O/OSA deployment. The RPC-O deployment script skips the
+  repo-build playbook to save time, but changes in the repo-build process in
+  OSA mean that if the repo0build playbook is run it will skip all the build
+  processes anyway as long as none of the requirements have changed.
+
+* Container artifact consumption is implemented through the use of the
+  ``lxc_container_variant`` variable set for each group in group vars, and
+  various global settings implemented in ``group_vars/all/lxc.yml``. These
+  settings instruct LXC to use the RPC-O prepared chroots when building the
+  container instead of using the upstream LXC container base.
+
```

Comment:
To check if your container is properly coming from an container build artefact, check for the presence of the proper code available + ansible facts in /etc/ansible/facts.d/.

Reply:
The same will be present in a non-artifacted build. Also, this is not meant to be an operators manual - only an orientation.

Reply:
fair enough.